### ISSUE TITLE
Import: Allow configuration of a custom file path pattern

### ISSUE THREAD

**Issue by lastzero** [photoprism/photoprism#4666]
**As an admin using the import functionality, I want to be able to configure the destination file path naming pattern so that I don't have to stick with the default if it doesn't work for me.**

In a first step, this will allow you to manually set a custom naming pattern in the [`settings.yml` config file](https://docs.photoprism.app/getting-started/config-files/settings/#media-library), e.g:

```yaml
Import:
  Dest: 2006/01/20060102_150405_82F63B78.jpg
```

The date and time placeholders for this are described at https://pkg.go.dev/time#Layout. `82F63B78` is a placeholder for the CRC32 checksum and `.jpg` for the file extension. Using a different 8 character hex number for the checksum and e.g. `.ext` for the extension will also work.

Once we consider this feature stable enough, we may add the ability to configure the naming pattern from the user interface, so that less experienced users can change it.

**Comment by lastzero** [comment]
Note that this new feature is currently only available in the `upgrade/vuetify-3` branch, so it cannot be tested on our public demo or with the `photoprism/photoprism:preview` image yet:
- https://github.com/photoprism/photoprism/tree/upgrade/vuetify-3

It will be available for public testing once we merge this feature branch into the `develop` branch in the coming weeks.

**Comment by maximecurioni** [comment]
@lastzero on the topic of controlling the destination filename, one common problem that I have run into over the years using Photoprism is that an import will make a particular original filename fixed, regardless of the date/time updates I may apply.

As an example, tonight, I just imported photos from 2022 and I found out after the import that they didn't have any timestamp information for Photoprism to parse. All these files are now stored under `originals/2025/02`.

I wrote an API script to fix the dates, using the file creation information I still had available on my machine. While Photoprism now correctly prints the updated date/time for these files after I successfully ran the script, there are still stored under `originals/2025/02`.

In a perfect world, I would have the option for any date/time update to automatically trigger a rename of the original filename to its new "expected" filename. That rename would be idempotent to future re-indexing.

Is this something that I can do using Photoprism's API (i.e. force the rename of the original filename)?

Of course, I  understand that we would potentially need to update EXIF data / video metadata to keep everything consistent (like https://github.com/photoprism/photoprism-contrib/blob/main/scripts/YAML%20to%20EXIF/README.md), but I would definitely feel it would be an improvement overall.

**Comment by lastzero** [comment]
@maximecurioni We appreciate your feedback! While configurable import folders will be available in our upcoming release, it will not (yet) allow you to rename files that have already been imported, as this may cause conflicts with other tools or instances that may be accessing your files. In addition, renaming files may cause storage and/or transfer overhead with backup tools that do not recognize that files have been moved, i.e. they may create and/or transfer a new backup copy.

**Comment by robertDpi** [comment]
Import:
  Path: /
  Move: false
  Dest: 2006/2006-01-02/{{.OriginalName}}

would this format work?
No matter what changes i do in the time format, Photoprism still imports in the same format. E.g. 
2017/12/renamed_filename_with_checksum.jpg

**Comment by lastzero** [comment]
No, using the original name as import name is not supported as that would require conflict resolution in case the name already exists. Also, it would not be possible to see if a file was imported by looking at the filename. However, the original name is recorded in the database and thus not lost.

**Comment by robertDpi** [comment]
Thank you for your reply. 
Conflict resolution could be an interesting feature. User may set preferences such as:
- Skip file
- Overwrite
- Rename

For me, keeping the original name is critical to guarantee consistency between backups, where filename is used to differentiate the files. 

Just to clarify, If I manually import some pictures in Photoprism originals folders, i need to perform a complete re-index in order to see them in the library?

**Comment by lastzero** [comment]
A regular rescan, which will add only the new files, will do. When you use WebDAV, the index can be updated automatically (see config options). If the original name is important for your backups, I recommend to transfer the files directly  to originals and not to use the import feature.
